metadata-ingestion/src/datahub/ingestion/source/powerbi/rest_api_wrapper/powerbi_api.py

```diff
@@ -0,0 +1,341 @@
+import json
+import logging
+from typing import Any, Dict, List, Optional
+
+import requests
+
+from datahub.ingestion.source.powerbi.config import (
+    PowerBiDashboardSourceConfig,
+    PowerBiDashboardSourceReport,
+)
+from datahub.ingestion.source.powerbi.rest_api_wrapper import data_resolver
+from datahub.ingestion.source.powerbi.rest_api_wrapper.data_classes import (
+    Dashboard,
+    PowerBIDataset,
+    Report,
+    Table,
+    User,
+    Workspace,
+)
+from datahub.ingestion.source.powerbi.rest_api_wrapper.data_resolver import (
+    AdminAPIResolver,
+    RegularAPIResolver,
+)
+
+# Logger instance
+logger = logging.getLogger(__name__)
+
+
+class PowerBiAPI:
+    def __init__(self, config: PowerBiDashboardSourceConfig) -> None:
+        self.__config: PowerBiDashboardSourceConfig = config
+
+        self.__regular_api_resolver = RegularAPIResolver(
+            client_id=self.__config.client_id,
+            client_secret=self.__config.client_secret,
+            tenant_id=self.__config.tenant_id,
+        )
+
+        self.__admin_api_resolver = AdminAPIResolver(
+            client_id=self.__config.client_id,
+            client_secret=self.__config.client_secret,
+            tenant_id=self.__config.tenant_id,
+        )
+
+    def _get_dashboard_endorsements(
+        self, scan_result: Optional[dict]
+    ) -> Dict[str, List[str]]:
+        """
+        Store saved dashboard endorsements into a dict with dashboard id as key and
+        endorsements or tags as list of strings
+        """
+        results: Dict[str, List[str]] = {}
+        if scan_result is None:
+            return results
+
+        for scanned_dashboard in scan_result["dashboards"]:
```

Comment:
Will scan_result always contain a key called dashboards? 

If not, what will happen here???

Reply:
replaced with scan_result.get(Constant.DASHBOARDS, [])